hibernate-core/src/main/java/org/hibernate/engine/internal/TwoPhaseLoad.java

```diff
@@ -288,15 +296,78 @@ else if ( value!= PropertyAccessStrategyBackRefImpl.UNKNOWN ) {
 			factory.getStatistics().loadEntity( persister.getEntityName() );
 		}
 	}
-	
+
+	/**
+	 * Check if eager of the association is overriden by anything.
+	 *
+	 * @param session
+	 * @param entityName
+	 * @param associationName
+	 *
+	 * @return null if there is no overriding, true if it is overridden to eager and false if it is overridden to lazy
+	 */
+	private static Boolean getOverridingEager(
+			SharedSessionContractImplementor session,
+			String entityName,
+			String associationName,
+			Type type) {
+		if ( type.isAssociationType() || type.isCollectionType() ) {
+			return getOverridingEagerFromFetchProfile( session, entityName, associationName );
+		}
+		return null;
+	}
+
+	private static Boolean getOverridingEagerFromFetchProfile(
+			SharedSessionContractImplementor session,
+			String entityName,
+			String associationName) {
+
+		Boolean result = null;
+		for ( FetchProfile fp : getActiveFetchProfiles( session ) ) {
+			Fetch f = fp.getFetchByRole( entityName + "." + associationName );
+			if ( f != null ) {
+				result = isEagerFetching( f );
+				break;
+			}
+		}
+
+		if ( LOG.isDebugEnabled() ) {
+			if ( result != null ) {
+				LOG.debugf(
+						"Overriding eager by active fetch profiles. EntityName: %s, associationName: %s, result: %s",
+						entityName,
+						associationName,
+						result
+				);
+			}
+		}
+
+		return result;
+	}
+
+	private static Collection<FetchProfile> getActiveFetchProfiles(SharedSessionContractImplementor session) {
+		ArrayList<FetchProfile> result = new ArrayList<>();
+		LoadQueryInfluencers loadQueryInfluencers = session.getLoadQueryInfluencers();
+		for ( String fetchProfileName : loadQueryInfluencers.getEnabledFetchProfileNames() ) {
+			FetchProfile fp = session.getFactory().getFetchProfile( fetchProfileName );
+			result.add( fp );
+		}
+		return result;
+	}
+
+	private static boolean isEagerFetching(Fetch fetch) {
+		Fetch.Style style = fetch.getStyle();
+		return style == Fetch.Style.JOIN || style == Fetch.Style.SELECT;
```

Comment:
FetchStyle.SELECT isn't associated to `FetchType.LAZY`?

Reply:
I am not sure I understand the question but generally this method decides if @FetchOverride section of Fetch Profile should cause eager loading or not.
Actually the method does not have to be here because org.hibernate.engine.profile.Fetch.Style has only two possible values - JOIN and SELECT, there is nothing like LAZY, so it always cause eager fetching, it is impossible now to change eager association to lazy by Fetch Profile. Maybe in the future?

Reply:
EAGER can be overridden with Hibernate. Only Lazy can. For FetchProfile, SELECT is not even supported.